From 87016281ac0a349f8a1cc456ea20d4932b780387 Mon Sep 17 00:00:00 2001
From: Szabolcs Nagy <szabolcs.nagy@arm.com>
Date: Fri, 26 Jan 2018 16:34:22 +0000
Subject: [PATCH 09/10] aarch64: Update ILP32 abilists

Update libc and libm abilists with copy_file_range, mlock2, pkey_* and
_FloatN related symbols. Remove pthread_self from the libpthread abilist.
This will need to be merged with the initial ilp32 abilist commit.

2018-01-26  Szabolcs Nagy  <szabolcs.nagy@arm.com>

	* sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist: Update.
	* sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist: Update.
	* sysdeps/unix/sysv/linux/aarch64/ilp32/libpthread.abilist: Update.
---
 sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist |  22 ++
 sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist | 312 +++++++++++++++++++++
 .../sysv/linux/aarch64/ilp32/libpthread.abilist    |   1 -
 3 files changed, 334 insertions(+), 1 deletion(-)

diff --git a/sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist b/sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist
index dd0ed329ab..55d2561bb5 100644
--- a/sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist
+++ b/sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist
@@ -645,6 +645,7 @@ GLIBC_2.27 closedir F
 GLIBC_2.27 closelog F
 GLIBC_2.27 confstr F
 GLIBC_2.27 connect F
+GLIBC_2.27 copy_file_range F
 GLIBC_2.27 copysign F
 GLIBC_2.27 copysignf F
 GLIBC_2.27 copysignl F
@@ -1271,6 +1272,7 @@ GLIBC_2.27 mkstemps64 F
 GLIBC_2.27 mktemp F
 GLIBC_2.27 mktime F
 GLIBC_2.27 mlock F
+GLIBC_2.27 mlock2 F
 GLIBC_2.27 mlockall F
 GLIBC_2.27 mmap F
 GLIBC_2.27 mmap64 F
@@ -1343,6 +1345,11 @@ GLIBC_2.27 personality F
 GLIBC_2.27 pipe F
 GLIBC_2.27 pipe2 F
 GLIBC_2.27 pivot_root F
+GLIBC_2.27 pkey_alloc F
+GLIBC_2.27 pkey_free F
+GLIBC_2.27 pkey_get F
+GLIBC_2.27 pkey_mprotect F
+GLIBC_2.27 pkey_set F
 GLIBC_2.27 pmap_getmaps F
 GLIBC_2.27 pmap_getport F
 GLIBC_2.27 pmap_rmtcall F
@@ -1712,6 +1719,9 @@ GLIBC_2.27 strfmon_l F
 GLIBC_2.27 strfromd F
 GLIBC_2.27 strfromf F
 GLIBC_2.27 strfromf128 F
+GLIBC_2.27 strfromf32 F
+GLIBC_2.27 strfromf32x F
+GLIBC_2.27 strfromf64 F
 GLIBC_2.27 strfromf64x F
 GLIBC_2.27 strfroml F
 GLIBC_2.27 strfry F
@@ -1738,6 +1748,12 @@ GLIBC_2.27 strtod_l F
 GLIBC_2.27 strtof F
 GLIBC_2.27 strtof128 F
 GLIBC_2.27 strtof128_l F
+GLIBC_2.27 strtof32 F
+GLIBC_2.27 strtof32_l F
+GLIBC_2.27 strtof32x F
+GLIBC_2.27 strtof32x_l F
+GLIBC_2.27 strtof64 F
+GLIBC_2.27 strtof64_l F
 GLIBC_2.27 strtof64x F
 GLIBC_2.27 strtof64x_l F
 GLIBC_2.27 strtof_l F
@@ -1955,6 +1971,12 @@ GLIBC_2.27 wcstod_l F
 GLIBC_2.27 wcstof F
 GLIBC_2.27 wcstof128 F
 GLIBC_2.27 wcstof128_l F
+GLIBC_2.27 wcstof32 F
+GLIBC_2.27 wcstof32_l F
+GLIBC_2.27 wcstof32x F
+GLIBC_2.27 wcstof32x_l F
+GLIBC_2.27 wcstof64 F
+GLIBC_2.27 wcstof64_l F
 GLIBC_2.27 wcstof64x F
 GLIBC_2.27 wcstof64x_l F
 GLIBC_2.27 wcstof_l F
diff --git a/sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist b/sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist
index 51a53dafd1..971af52586 100644
--- a/sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist
+++ b/sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist
@@ -102,20 +102,32 @@ GLIBC_2.27 __ynl_finite F
 GLIBC_2.27 acos F
 GLIBC_2.27 acosf F
 GLIBC_2.27 acosf128 F
+GLIBC_2.27 acosf32 F
+GLIBC_2.27 acosf32x F
+GLIBC_2.27 acosf64 F
 GLIBC_2.27 acosf64x F
 GLIBC_2.27 acosh F
 GLIBC_2.27 acoshf F
 GLIBC_2.27 acoshf128 F
+GLIBC_2.27 acoshf32 F
+GLIBC_2.27 acoshf32x F
+GLIBC_2.27 acoshf64 F
 GLIBC_2.27 acoshf64x F
 GLIBC_2.27 acoshl F
 GLIBC_2.27 acosl F
 GLIBC_2.27 asin F
 GLIBC_2.27 asinf F
 GLIBC_2.27 asinf128 F
+GLIBC_2.27 asinf32 F
+GLIBC_2.27 asinf32x F
+GLIBC_2.27 asinf64 F
 GLIBC_2.27 asinf64x F
 GLIBC_2.27 asinh F
 GLIBC_2.27 asinhf F
 GLIBC_2.27 asinhf128 F
+GLIBC_2.27 asinhf32 F
+GLIBC_2.27 asinhf32x F
+GLIBC_2.27 asinhf64 F
 GLIBC_2.27 asinhf64x F
 GLIBC_2.27 asinhl F
 GLIBC_2.27 asinl F
@@ -123,159 +135,255 @@ GLIBC_2.27 atan F
 GLIBC_2.27 atan2 F
 GLIBC_2.27 atan2f F
 GLIBC_2.27 atan2f128 F
+GLIBC_2.27 atan2f32 F
+GLIBC_2.27 atan2f32x F
+GLIBC_2.27 atan2f64 F
 GLIBC_2.27 atan2f64x F
 GLIBC_2.27 atan2l F
 GLIBC_2.27 atanf F
 GLIBC_2.27 atanf128 F
+GLIBC_2.27 atanf32 F
+GLIBC_2.27 atanf32x F
+GLIBC_2.27 atanf64 F
 GLIBC_2.27 atanf64x F
 GLIBC_2.27 atanh F
 GLIBC_2.27 atanhf F
 GLIBC_2.27 atanhf128 F
+GLIBC_2.27 atanhf32 F
+GLIBC_2.27 atanhf32x F
+GLIBC_2.27 atanhf64 F
 GLIBC_2.27 atanhf64x F
 GLIBC_2.27 atanhl F
 GLIBC_2.27 atanl F
 GLIBC_2.27 cabs F
 GLIBC_2.27 cabsf F
 GLIBC_2.27 cabsf128 F
+GLIBC_2.27 cabsf32 F
+GLIBC_2.27 cabsf32x F
+GLIBC_2.27 cabsf64 F
 GLIBC_2.27 cabsf64x F
 GLIBC_2.27 cabsl F
 GLIBC_2.27 cacos F
 GLIBC_2.27 cacosf F
 GLIBC_2.27 cacosf128 F
+GLIBC_2.27 cacosf32 F
+GLIBC_2.27 cacosf32x F
+GLIBC_2.27 cacosf64 F
 GLIBC_2.27 cacosf64x F
 GLIBC_2.27 cacosh F
 GLIBC_2.27 cacoshf F
 GLIBC_2.27 cacoshf128 F
+GLIBC_2.27 cacoshf32 F
+GLIBC_2.27 cacoshf32x F
+GLIBC_2.27 cacoshf64 F
 GLIBC_2.27 cacoshf64x F
 GLIBC_2.27 cacoshl F
 GLIBC_2.27 cacosl F
 GLIBC_2.27 canonicalize F
 GLIBC_2.27 canonicalizef F
 GLIBC_2.27 canonicalizef128 F
+GLIBC_2.27 canonicalizef32 F
+GLIBC_2.27 canonicalizef32x F
+GLIBC_2.27 canonicalizef64 F
 GLIBC_2.27 canonicalizef64x F
 GLIBC_2.27 canonicalizel F
 GLIBC_2.27 carg F
 GLIBC_2.27 cargf F
 GLIBC_2.27 cargf128 F
+GLIBC_2.27 cargf32 F
+GLIBC_2.27 cargf32x F
+GLIBC_2.27 cargf64 F
 GLIBC_2.27 cargf64x F
 GLIBC_2.27 cargl F
 GLIBC_2.27 casin F
 GLIBC_2.27 casinf F
 GLIBC_2.27 casinf128 F
+GLIBC_2.27 casinf32 F
+GLIBC_2.27 casinf32x F
+GLIBC_2.27 casinf64 F
 GLIBC_2.27 casinf64x F
 GLIBC_2.27 casinh F
 GLIBC_2.27 casinhf F
 GLIBC_2.27 casinhf128 F
+GLIBC_2.27 casinhf32 F
+GLIBC_2.27 casinhf32x F
+GLIBC_2.27 casinhf64 F
 GLIBC_2.27 casinhf64x F
 GLIBC_2.27 casinhl F
 GLIBC_2.27 casinl F
 GLIBC_2.27 catan F
 GLIBC_2.27 catanf F
 GLIBC_2.27 catanf128 F
+GLIBC_2.27 catanf32 F
+GLIBC_2.27 catanf32x F
+GLIBC_2.27 catanf64 F
 GLIBC_2.27 catanf64x F
 GLIBC_2.27 catanh F
 GLIBC_2.27 catanhf F
 GLIBC_2.27 catanhf128 F
+GLIBC_2.27 catanhf32 F
+GLIBC_2.27 catanhf32x F
+GLIBC_2.27 catanhf64 F
 GLIBC_2.27 catanhf64x F
 GLIBC_2.27 catanhl F
 GLIBC_2.27 catanl F
 GLIBC_2.27 cbrt F
 GLIBC_2.27 cbrtf F
 GLIBC_2.27 cbrtf128 F
+GLIBC_2.27 cbrtf32 F
+GLIBC_2.27 cbrtf32x F
+GLIBC_2.27 cbrtf64 F
 GLIBC_2.27 cbrtf64x F
 GLIBC_2.27 cbrtl F
 GLIBC_2.27 ccos F
 GLIBC_2.27 ccosf F
 GLIBC_2.27 ccosf128 F
+GLIBC_2.27 ccosf32 F
+GLIBC_2.27 ccosf32x F
+GLIBC_2.27 ccosf64 F
 GLIBC_2.27 ccosf64x F
 GLIBC_2.27 ccosh F
 GLIBC_2.27 ccoshf F
 GLIBC_2.27 ccoshf128 F
+GLIBC_2.27 ccoshf32 F
+GLIBC_2.27 ccoshf32x F
+GLIBC_2.27 ccoshf64 F
 GLIBC_2.27 ccoshf64x F
 GLIBC_2.27 ccoshl F
 GLIBC_2.27 ccosl F
 GLIBC_2.27 ceil F
 GLIBC_2.27 ceilf F
 GLIBC_2.27 ceilf128 F
+GLIBC_2.27 ceilf32 F
+GLIBC_2.27 ceilf32x F
+GLIBC_2.27 ceilf64 F
 GLIBC_2.27 ceilf64x F
 GLIBC_2.27 ceill F
 GLIBC_2.27 cexp F
 GLIBC_2.27 cexpf F
 GLIBC_2.27 cexpf128 F
+GLIBC_2.27 cexpf32 F
+GLIBC_2.27 cexpf32x F
+GLIBC_2.27 cexpf64 F
 GLIBC_2.27 cexpf64x F
 GLIBC_2.27 cexpl F
 GLIBC_2.27 cimag F
 GLIBC_2.27 cimagf F
 GLIBC_2.27 cimagf128 F
+GLIBC_2.27 cimagf32 F
+GLIBC_2.27 cimagf32x F
+GLIBC_2.27 cimagf64 F
 GLIBC_2.27 cimagf64x F
 GLIBC_2.27 cimagl F
 GLIBC_2.27 clog F
 GLIBC_2.27 clog10 F
 GLIBC_2.27 clog10f F
 GLIBC_2.27 clog10f128 F
+GLIBC_2.27 clog10f32 F
+GLIBC_2.27 clog10f32x F
+GLIBC_2.27 clog10f64 F
 GLIBC_2.27 clog10f64x F
 GLIBC_2.27 clog10l F
 GLIBC_2.27 clogf F
 GLIBC_2.27 clogf128 F
+GLIBC_2.27 clogf32 F
+GLIBC_2.27 clogf32x F
+GLIBC_2.27 clogf64 F
 GLIBC_2.27 clogf64x F
 GLIBC_2.27 clogl F
 GLIBC_2.27 conj F
 GLIBC_2.27 conjf F
 GLIBC_2.27 conjf128 F
+GLIBC_2.27 conjf32 F
+GLIBC_2.27 conjf32x F
+GLIBC_2.27 conjf64 F
 GLIBC_2.27 conjf64x F
 GLIBC_2.27 conjl F
 GLIBC_2.27 copysign F
 GLIBC_2.27 copysignf F
 GLIBC_2.27 copysignf128 F
+GLIBC_2.27 copysignf32 F
+GLIBC_2.27 copysignf32x F
+GLIBC_2.27 copysignf64 F
 GLIBC_2.27 copysignf64x F
 GLIBC_2.27 copysignl F
 GLIBC_2.27 cos F
 GLIBC_2.27 cosf F
 GLIBC_2.27 cosf128 F
+GLIBC_2.27 cosf32 F
+GLIBC_2.27 cosf32x F
+GLIBC_2.27 cosf64 F
 GLIBC_2.27 cosf64x F
 GLIBC_2.27 cosh F
 GLIBC_2.27 coshf F
 GLIBC_2.27 coshf128 F
+GLIBC_2.27 coshf32 F
+GLIBC_2.27 coshf32x F
+GLIBC_2.27 coshf64 F
 GLIBC_2.27 coshf64x F
 GLIBC_2.27 coshl F
 GLIBC_2.27 cosl F
 GLIBC_2.27 cpow F
 GLIBC_2.27 cpowf F
 GLIBC_2.27 cpowf128 F
+GLIBC_2.27 cpowf32 F
+GLIBC_2.27 cpowf32x F
+GLIBC_2.27 cpowf64 F
 GLIBC_2.27 cpowf64x F
 GLIBC_2.27 cpowl F
 GLIBC_2.27 cproj F
 GLIBC_2.27 cprojf F
 GLIBC_2.27 cprojf128 F
+GLIBC_2.27 cprojf32 F
+GLIBC_2.27 cprojf32x F
+GLIBC_2.27 cprojf64 F
 GLIBC_2.27 cprojf64x F
 GLIBC_2.27 cprojl F
 GLIBC_2.27 creal F
 GLIBC_2.27 crealf F
 GLIBC_2.27 crealf128 F
+GLIBC_2.27 crealf32 F
+GLIBC_2.27 crealf32x F
+GLIBC_2.27 crealf64 F
 GLIBC_2.27 crealf64x F
 GLIBC_2.27 creall F
 GLIBC_2.27 csin F
 GLIBC_2.27 csinf F
 GLIBC_2.27 csinf128 F
+GLIBC_2.27 csinf32 F
+GLIBC_2.27 csinf32x F
+GLIBC_2.27 csinf64 F
 GLIBC_2.27 csinf64x F
 GLIBC_2.27 csinh F
 GLIBC_2.27 csinhf F
 GLIBC_2.27 csinhf128 F
+GLIBC_2.27 csinhf32 F
+GLIBC_2.27 csinhf32x F
+GLIBC_2.27 csinhf64 F
 GLIBC_2.27 csinhf64x F
 GLIBC_2.27 csinhl F
 GLIBC_2.27 csinl F
 GLIBC_2.27 csqrt F
 GLIBC_2.27 csqrtf F
 GLIBC_2.27 csqrtf128 F
+GLIBC_2.27 csqrtf32 F
+GLIBC_2.27 csqrtf32x F
+GLIBC_2.27 csqrtf64 F
 GLIBC_2.27 csqrtf64x F
 GLIBC_2.27 csqrtl F
 GLIBC_2.27 ctan F
 GLIBC_2.27 ctanf F
 GLIBC_2.27 ctanf128 F
+GLIBC_2.27 ctanf32 F
+GLIBC_2.27 ctanf32x F
+GLIBC_2.27 ctanf64 F
 GLIBC_2.27 ctanf64x F
 GLIBC_2.27 ctanh F
 GLIBC_2.27 ctanhf F
 GLIBC_2.27 ctanhf128 F
+GLIBC_2.27 ctanhf32 F
+GLIBC_2.27 ctanhf32x F
+GLIBC_2.27 ctanhf64 F
 GLIBC_2.27 ctanhf64x F
 GLIBC_2.27 ctanhl F
 GLIBC_2.27 ctanl F
@@ -286,40 +394,64 @@ GLIBC_2.27 erf F
 GLIBC_2.27 erfc F
 GLIBC_2.27 erfcf F
 GLIBC_2.27 erfcf128 F
+GLIBC_2.27 erfcf32 F
+GLIBC_2.27 erfcf32x F
+GLIBC_2.27 erfcf64 F
 GLIBC_2.27 erfcf64x F
 GLIBC_2.27 erfcl F
 GLIBC_2.27 erff F
 GLIBC_2.27 erff128 F
+GLIBC_2.27 erff32 F
+GLIBC_2.27 erff32x F
+GLIBC_2.27 erff64 F
 GLIBC_2.27 erff64x F
 GLIBC_2.27 erfl F
 GLIBC_2.27 exp F
 GLIBC_2.27 exp10 F
 GLIBC_2.27 exp10f F
 GLIBC_2.27 exp10f128 F
+GLIBC_2.27 exp10f32 F
+GLIBC_2.27 exp10f32x F
+GLIBC_2.27 exp10f64 F
 GLIBC_2.27 exp10f64x F
 GLIBC_2.27 exp10l F
 GLIBC_2.27 exp2 F
 GLIBC_2.27 exp2f F
 GLIBC_2.27 exp2f128 F
+GLIBC_2.27 exp2f32 F
+GLIBC_2.27 exp2f32x F
+GLIBC_2.27 exp2f64 F
 GLIBC_2.27 exp2f64x F
 GLIBC_2.27 exp2l F
 GLIBC_2.27 expf F
 GLIBC_2.27 expf128 F
+GLIBC_2.27 expf32 F
+GLIBC_2.27 expf32x F
+GLIBC_2.27 expf64 F
 GLIBC_2.27 expf64x F
 GLIBC_2.27 expl F
 GLIBC_2.27 expm1 F
 GLIBC_2.27 expm1f F
 GLIBC_2.27 expm1f128 F
+GLIBC_2.27 expm1f32 F
+GLIBC_2.27 expm1f32x F
+GLIBC_2.27 expm1f64 F
 GLIBC_2.27 expm1f64x F
 GLIBC_2.27 expm1l F
 GLIBC_2.27 fabs F
 GLIBC_2.27 fabsf F
 GLIBC_2.27 fabsf128 F
+GLIBC_2.27 fabsf32 F
+GLIBC_2.27 fabsf32x F
+GLIBC_2.27 fabsf64 F
 GLIBC_2.27 fabsf64x F
 GLIBC_2.27 fabsl F
 GLIBC_2.27 fdim F
 GLIBC_2.27 fdimf F
 GLIBC_2.27 fdimf128 F
+GLIBC_2.27 fdimf32 F
+GLIBC_2.27 fdimf32x F
+GLIBC_2.27 fdimf64 F
 GLIBC_2.27 fdimf64x F
 GLIBC_2.27 fdiml F
 GLIBC_2.27 feclearexcept F
@@ -346,51 +478,81 @@ GLIBC_2.27 finitel F
 GLIBC_2.27 floor F
 GLIBC_2.27 floorf F
 GLIBC_2.27 floorf128 F
+GLIBC_2.27 floorf32 F
+GLIBC_2.27 floorf32x F
+GLIBC_2.27 floorf64 F
 GLIBC_2.27 floorf64x F
 GLIBC_2.27 floorl F
 GLIBC_2.27 fma F
 GLIBC_2.27 fmaf F
 GLIBC_2.27 fmaf128 F
+GLIBC_2.27 fmaf32 F
+GLIBC_2.27 fmaf32x F
+GLIBC_2.27 fmaf64 F
 GLIBC_2.27 fmaf64x F
 GLIBC_2.27 fmal F
 GLIBC_2.27 fmax F
 GLIBC_2.27 fmaxf F
 GLIBC_2.27 fmaxf128 F
+GLIBC_2.27 fmaxf32 F
+GLIBC_2.27 fmaxf32x F
+GLIBC_2.27 fmaxf64 F
 GLIBC_2.27 fmaxf64x F
 GLIBC_2.27 fmaxl F
 GLIBC_2.27 fmaxmag F
 GLIBC_2.27 fmaxmagf F
 GLIBC_2.27 fmaxmagf128 F
+GLIBC_2.27 fmaxmagf32 F
+GLIBC_2.27 fmaxmagf32x F
+GLIBC_2.27 fmaxmagf64 F
 GLIBC_2.27 fmaxmagf64x F
 GLIBC_2.27 fmaxmagl F
 GLIBC_2.27 fmin F
 GLIBC_2.27 fminf F
 GLIBC_2.27 fminf128 F
+GLIBC_2.27 fminf32 F
+GLIBC_2.27 fminf32x F
+GLIBC_2.27 fminf64 F
 GLIBC_2.27 fminf64x F
 GLIBC_2.27 fminl F
 GLIBC_2.27 fminmag F
 GLIBC_2.27 fminmagf F
 GLIBC_2.27 fminmagf128 F
+GLIBC_2.27 fminmagf32 F
+GLIBC_2.27 fminmagf32x F
+GLIBC_2.27 fminmagf64 F
 GLIBC_2.27 fminmagf64x F
 GLIBC_2.27 fminmagl F
 GLIBC_2.27 fmod F
 GLIBC_2.27 fmodf F
 GLIBC_2.27 fmodf128 F
+GLIBC_2.27 fmodf32 F
+GLIBC_2.27 fmodf32x F
+GLIBC_2.27 fmodf64 F
 GLIBC_2.27 fmodf64x F
 GLIBC_2.27 fmodl F
 GLIBC_2.27 frexp F
 GLIBC_2.27 frexpf F
 GLIBC_2.27 frexpf128 F
+GLIBC_2.27 frexpf32 F
+GLIBC_2.27 frexpf32x F
+GLIBC_2.27 frexpf64 F
 GLIBC_2.27 frexpf64x F
 GLIBC_2.27 frexpl F
 GLIBC_2.27 fromfp F
 GLIBC_2.27 fromfpf F
 GLIBC_2.27 fromfpf128 F
+GLIBC_2.27 fromfpf32 F
+GLIBC_2.27 fromfpf32x F
+GLIBC_2.27 fromfpf64 F
 GLIBC_2.27 fromfpf64x F
 GLIBC_2.27 fromfpl F
 GLIBC_2.27 fromfpx F
 GLIBC_2.27 fromfpxf F
 GLIBC_2.27 fromfpxf128 F
+GLIBC_2.27 fromfpxf32 F
+GLIBC_2.27 fromfpxf32x F
+GLIBC_2.27 fromfpxf64 F
 GLIBC_2.27 fromfpxf64x F
 GLIBC_2.27 fromfpxl F
 GLIBC_2.27 gamma F
@@ -399,36 +561,57 @@ GLIBC_2.27 gammal F
 GLIBC_2.27 getpayload F
 GLIBC_2.27 getpayloadf F
 GLIBC_2.27 getpayloadf128 F
+GLIBC_2.27 getpayloadf32 F
+GLIBC_2.27 getpayloadf32x F
+GLIBC_2.27 getpayloadf64 F
 GLIBC_2.27 getpayloadf64x F
 GLIBC_2.27 getpayloadl F
 GLIBC_2.27 hypot F
 GLIBC_2.27 hypotf F
 GLIBC_2.27 hypotf128 F
+GLIBC_2.27 hypotf32 F
+GLIBC_2.27 hypotf32x F
+GLIBC_2.27 hypotf64 F
 GLIBC_2.27 hypotf64x F
 GLIBC_2.27 hypotl F
 GLIBC_2.27 ilogb F
 GLIBC_2.27 ilogbf F
 GLIBC_2.27 ilogbf128 F
+GLIBC_2.27 ilogbf32 F
+GLIBC_2.27 ilogbf32x F
+GLIBC_2.27 ilogbf64 F
 GLIBC_2.27 ilogbf64x F
 GLIBC_2.27 ilogbl F
 GLIBC_2.27 j0 F
 GLIBC_2.27 j0f F
 GLIBC_2.27 j0f128 F
+GLIBC_2.27 j0f32 F
+GLIBC_2.27 j0f32x F
+GLIBC_2.27 j0f64 F
 GLIBC_2.27 j0f64x F
 GLIBC_2.27 j0l F
 GLIBC_2.27 j1 F
 GLIBC_2.27 j1f F
 GLIBC_2.27 j1f128 F
+GLIBC_2.27 j1f32 F
+GLIBC_2.27 j1f32x F
+GLIBC_2.27 j1f64 F
 GLIBC_2.27 j1f64x F
 GLIBC_2.27 j1l F
 GLIBC_2.27 jn F
 GLIBC_2.27 jnf F
 GLIBC_2.27 jnf128 F
+GLIBC_2.27 jnf32 F
+GLIBC_2.27 jnf32x F
+GLIBC_2.27 jnf64 F
 GLIBC_2.27 jnf64x F
 GLIBC_2.27 jnl F
 GLIBC_2.27 ldexp F
 GLIBC_2.27 ldexpf F
 GLIBC_2.27 ldexpf128 F
+GLIBC_2.27 ldexpf32 F
+GLIBC_2.27 ldexpf32x F
+GLIBC_2.27 ldexpf64 F
 GLIBC_2.27 ldexpf64x F
 GLIBC_2.27 ldexpl F
 GLIBC_2.27 lgamma F
@@ -436,6 +619,12 @@ GLIBC_2.27 lgamma_r F
 GLIBC_2.27 lgammaf F
 GLIBC_2.27 lgammaf128 F
 GLIBC_2.27 lgammaf128_r F
+GLIBC_2.27 lgammaf32 F
+GLIBC_2.27 lgammaf32_r F
+GLIBC_2.27 lgammaf32x F
+GLIBC_2.27 lgammaf32x_r F
+GLIBC_2.27 lgammaf64 F
+GLIBC_2.27 lgammaf64_r F
 GLIBC_2.27 lgammaf64x F
 GLIBC_2.27 lgammaf64x_r F
 GLIBC_2.27 lgammaf_r F
@@ -444,76 +633,121 @@ GLIBC_2.27 lgammal_r F
 GLIBC_2.27 llogb F
 GLIBC_2.27 llogbf F
 GLIBC_2.27 llogbf128 F
+GLIBC_2.27 llogbf32 F
+GLIBC_2.27 llogbf32x F
+GLIBC_2.27 llogbf64 F
 GLIBC_2.27 llogbf64x F
 GLIBC_2.27 llogbl F
 GLIBC_2.27 llrint F
 GLIBC_2.27 llrintf F
 GLIBC_2.27 llrintf128 F
+GLIBC_2.27 llrintf32 F
+GLIBC_2.27 llrintf32x F
+GLIBC_2.27 llrintf64 F
 GLIBC_2.27 llrintf64x F
 GLIBC_2.27 llrintl F
 GLIBC_2.27 llround F
 GLIBC_2.27 llroundf F
 GLIBC_2.27 llroundf128 F
+GLIBC_2.27 llroundf32 F
+GLIBC_2.27 llroundf32x F
+GLIBC_2.27 llroundf64 F
 GLIBC_2.27 llroundf64x F
 GLIBC_2.27 llroundl F
 GLIBC_2.27 log F
 GLIBC_2.27 log10 F
 GLIBC_2.27 log10f F
 GLIBC_2.27 log10f128 F
+GLIBC_2.27 log10f32 F
+GLIBC_2.27 log10f32x F
+GLIBC_2.27 log10f64 F
 GLIBC_2.27 log10f64x F
 GLIBC_2.27 log10l F
 GLIBC_2.27 log1p F
 GLIBC_2.27 log1pf F
 GLIBC_2.27 log1pf128 F
+GLIBC_2.27 log1pf32 F
+GLIBC_2.27 log1pf32x F
+GLIBC_2.27 log1pf64 F
 GLIBC_2.27 log1pf64x F
 GLIBC_2.27 log1pl F
 GLIBC_2.27 log2 F
 GLIBC_2.27 log2f F
 GLIBC_2.27 log2f128 F
+GLIBC_2.27 log2f32 F
+GLIBC_2.27 log2f32x F
+GLIBC_2.27 log2f64 F
 GLIBC_2.27 log2f64x F
 GLIBC_2.27 log2l F
 GLIBC_2.27 logb F
 GLIBC_2.27 logbf F
 GLIBC_2.27 logbf128 F
+GLIBC_2.27 logbf32 F
+GLIBC_2.27 logbf32x F
+GLIBC_2.27 logbf64 F
 GLIBC_2.27 logbf64x F
 GLIBC_2.27 logbl F
 GLIBC_2.27 logf F
 GLIBC_2.27 logf128 F
+GLIBC_2.27 logf32 F
+GLIBC_2.27 logf32x F
+GLIBC_2.27 logf64 F
 GLIBC_2.27 logf64x F
 GLIBC_2.27 logl F
 GLIBC_2.27 lrint F
 GLIBC_2.27 lrintf F
 GLIBC_2.27 lrintf128 F
+GLIBC_2.27 lrintf32 F
+GLIBC_2.27 lrintf32x F
+GLIBC_2.27 lrintf64 F
 GLIBC_2.27 lrintf64x F
 GLIBC_2.27 lrintl F
 GLIBC_2.27 lround F
 GLIBC_2.27 lroundf F
 GLIBC_2.27 lroundf128 F
+GLIBC_2.27 lroundf32 F
+GLIBC_2.27 lroundf32x F
+GLIBC_2.27 lroundf64 F
 GLIBC_2.27 lroundf64x F
 GLIBC_2.27 lroundl F
 GLIBC_2.27 modf F
 GLIBC_2.27 modff F
 GLIBC_2.27 modff128 F
+GLIBC_2.27 modff32 F
+GLIBC_2.27 modff32x F
+GLIBC_2.27 modff64 F
 GLIBC_2.27 modff64x F
 GLIBC_2.27 modfl F
 GLIBC_2.27 nan F
 GLIBC_2.27 nanf F
 GLIBC_2.27 nanf128 F
+GLIBC_2.27 nanf32 F
+GLIBC_2.27 nanf32x F
+GLIBC_2.27 nanf64 F
 GLIBC_2.27 nanf64x F
 GLIBC_2.27 nanl F
 GLIBC_2.27 nearbyint F
 GLIBC_2.27 nearbyintf F
 GLIBC_2.27 nearbyintf128 F
+GLIBC_2.27 nearbyintf32 F
+GLIBC_2.27 nearbyintf32x F
+GLIBC_2.27 nearbyintf64 F
 GLIBC_2.27 nearbyintf64x F
 GLIBC_2.27 nearbyintl F
 GLIBC_2.27 nextafter F
 GLIBC_2.27 nextafterf F
 GLIBC_2.27 nextafterf128 F
+GLIBC_2.27 nextafterf32 F
+GLIBC_2.27 nextafterf32x F
+GLIBC_2.27 nextafterf64 F
 GLIBC_2.27 nextafterf64x F
 GLIBC_2.27 nextafterl F
 GLIBC_2.27 nextdown F
 GLIBC_2.27 nextdownf F
 GLIBC_2.27 nextdownf128 F
+GLIBC_2.27 nextdownf32 F
+GLIBC_2.27 nextdownf32x F
+GLIBC_2.27 nextdownf64 F
 GLIBC_2.27 nextdownf64x F
 GLIBC_2.27 nextdownl F
 GLIBC_2.27 nexttoward F
@@ -522,36 +756,57 @@ GLIBC_2.27 nexttowardl F
 GLIBC_2.27 nextup F
 GLIBC_2.27 nextupf F
 GLIBC_2.27 nextupf128 F
+GLIBC_2.27 nextupf32 F
+GLIBC_2.27 nextupf32x F
+GLIBC_2.27 nextupf64 F
 GLIBC_2.27 nextupf64x F
 GLIBC_2.27 nextupl F
 GLIBC_2.27 pow F
 GLIBC_2.27 powf F
 GLIBC_2.27 powf128 F
+GLIBC_2.27 powf32 F
+GLIBC_2.27 powf32x F
+GLIBC_2.27 powf64 F
 GLIBC_2.27 powf64x F
 GLIBC_2.27 powl F
 GLIBC_2.27 remainder F
 GLIBC_2.27 remainderf F
 GLIBC_2.27 remainderf128 F
+GLIBC_2.27 remainderf32 F
+GLIBC_2.27 remainderf32x F
+GLIBC_2.27 remainderf64 F
 GLIBC_2.27 remainderf64x F
 GLIBC_2.27 remainderl F
 GLIBC_2.27 remquo F
 GLIBC_2.27 remquof F
 GLIBC_2.27 remquof128 F
+GLIBC_2.27 remquof32 F
+GLIBC_2.27 remquof32x F
+GLIBC_2.27 remquof64 F
 GLIBC_2.27 remquof64x F
 GLIBC_2.27 remquol F
 GLIBC_2.27 rint F
 GLIBC_2.27 rintf F
 GLIBC_2.27 rintf128 F
+GLIBC_2.27 rintf32 F
+GLIBC_2.27 rintf32x F
+GLIBC_2.27 rintf64 F
 GLIBC_2.27 rintf64x F
 GLIBC_2.27 rintl F
 GLIBC_2.27 round F
 GLIBC_2.27 roundeven F
 GLIBC_2.27 roundevenf F
 GLIBC_2.27 roundevenf128 F
+GLIBC_2.27 roundevenf32 F
+GLIBC_2.27 roundevenf32x F
+GLIBC_2.27 roundevenf64 F
 GLIBC_2.27 roundevenf64x F
 GLIBC_2.27 roundevenl F
 GLIBC_2.27 roundf F
 GLIBC_2.27 roundf128 F
+GLIBC_2.27 roundf32 F
+GLIBC_2.27 roundf32x F
+GLIBC_2.27 roundf64 F
 GLIBC_2.27 roundf64x F
 GLIBC_2.27 roundl F
 GLIBC_2.27 scalb F
@@ -560,21 +815,33 @@ GLIBC_2.27 scalbl F
 GLIBC_2.27 scalbln F
 GLIBC_2.27 scalblnf F
 GLIBC_2.27 scalblnf128 F
+GLIBC_2.27 scalblnf32 F
+GLIBC_2.27 scalblnf32x F
+GLIBC_2.27 scalblnf64 F
 GLIBC_2.27 scalblnf64x F
 GLIBC_2.27 scalblnl F
 GLIBC_2.27 scalbn F
 GLIBC_2.27 scalbnf F
 GLIBC_2.27 scalbnf128 F
+GLIBC_2.27 scalbnf32 F
+GLIBC_2.27 scalbnf32x F
+GLIBC_2.27 scalbnf64 F
 GLIBC_2.27 scalbnf64x F
 GLIBC_2.27 scalbnl F
 GLIBC_2.27 setpayload F
 GLIBC_2.27 setpayloadf F
 GLIBC_2.27 setpayloadf128 F
+GLIBC_2.27 setpayloadf32 F
+GLIBC_2.27 setpayloadf32x F
+GLIBC_2.27 setpayloadf64 F
 GLIBC_2.27 setpayloadf64x F
 GLIBC_2.27 setpayloadl F
 GLIBC_2.27 setpayloadsig F
 GLIBC_2.27 setpayloadsigf F
 GLIBC_2.27 setpayloadsigf128 F
+GLIBC_2.27 setpayloadsigf32 F
+GLIBC_2.27 setpayloadsigf32x F
+GLIBC_2.27 setpayloadsigf64 F
 GLIBC_2.27 setpayloadsigf64x F
 GLIBC_2.27 setpayloadsigl F
 GLIBC_2.27 signgam D 0x4
@@ -585,74 +852,119 @@ GLIBC_2.27 sin F
 GLIBC_2.27 sincos F
 GLIBC_2.27 sincosf F
 GLIBC_2.27 sincosf128 F
+GLIBC_2.27 sincosf32 F
+GLIBC_2.27 sincosf32x F
+GLIBC_2.27 sincosf64 F
 GLIBC_2.27 sincosf64x F
 GLIBC_2.27 sincosl F
 GLIBC_2.27 sinf F
 GLIBC_2.27 sinf128 F
+GLIBC_2.27 sinf32 F
+GLIBC_2.27 sinf32x F
+GLIBC_2.27 sinf64 F
 GLIBC_2.27 sinf64x F
 GLIBC_2.27 sinh F
 GLIBC_2.27 sinhf F
 GLIBC_2.27 sinhf128 F
+GLIBC_2.27 sinhf32 F
+GLIBC_2.27 sinhf32x F
+GLIBC_2.27 sinhf64 F
 GLIBC_2.27 sinhf64x F
 GLIBC_2.27 sinhl F
 GLIBC_2.27 sinl F
 GLIBC_2.27 sqrt F
 GLIBC_2.27 sqrtf F
 GLIBC_2.27 sqrtf128 F
+GLIBC_2.27 sqrtf32 F
+GLIBC_2.27 sqrtf32x F
+GLIBC_2.27 sqrtf64 F
 GLIBC_2.27 sqrtf64x F
 GLIBC_2.27 sqrtl F
 GLIBC_2.27 tan F
 GLIBC_2.27 tanf F
 GLIBC_2.27 tanf128 F
+GLIBC_2.27 tanf32 F
+GLIBC_2.27 tanf32x F
+GLIBC_2.27 tanf64 F
 GLIBC_2.27 tanf64x F
 GLIBC_2.27 tanh F
 GLIBC_2.27 tanhf F
 GLIBC_2.27 tanhf128 F
+GLIBC_2.27 tanhf32 F
+GLIBC_2.27 tanhf32x F
+GLIBC_2.27 tanhf64 F
 GLIBC_2.27 tanhf64x F
 GLIBC_2.27 tanhl F
 GLIBC_2.27 tanl F
 GLIBC_2.27 tgamma F
 GLIBC_2.27 tgammaf F
 GLIBC_2.27 tgammaf128 F
+GLIBC_2.27 tgammaf32 F
+GLIBC_2.27 tgammaf32x F
+GLIBC_2.27 tgammaf64 F
 GLIBC_2.27 tgammaf64x F
 GLIBC_2.27 tgammal F
 GLIBC_2.27 totalorder F
 GLIBC_2.27 totalorderf F
 GLIBC_2.27 totalorderf128 F
+GLIBC_2.27 totalorderf32 F
+GLIBC_2.27 totalorderf32x F
+GLIBC_2.27 totalorderf64 F
 GLIBC_2.27 totalorderf64x F
 GLIBC_2.27 totalorderl F
 GLIBC_2.27 totalordermag F
 GLIBC_2.27 totalordermagf F
 GLIBC_2.27 totalordermagf128 F
+GLIBC_2.27 totalordermagf32 F
+GLIBC_2.27 totalordermagf32x F
+GLIBC_2.27 totalordermagf64 F
 GLIBC_2.27 totalordermagf64x F
 GLIBC_2.27 totalordermagl F
 GLIBC_2.27 trunc F
 GLIBC_2.27 truncf F
 GLIBC_2.27 truncf128 F
+GLIBC_2.27 truncf32 F
+GLIBC_2.27 truncf32x F
+GLIBC_2.27 truncf64 F
 GLIBC_2.27 truncf64x F
 GLIBC_2.27 truncl F
 GLIBC_2.27 ufromfp F
 GLIBC_2.27 ufromfpf F
 GLIBC_2.27 ufromfpf128 F
+GLIBC_2.27 ufromfpf32 F
+GLIBC_2.27 ufromfpf32x F
+GLIBC_2.27 ufromfpf64 F
 GLIBC_2.27 ufromfpf64x F
 GLIBC_2.27 ufromfpl F
 GLIBC_2.27 ufromfpx F
 GLIBC_2.27 ufromfpxf F
 GLIBC_2.27 ufromfpxf128 F
+GLIBC_2.27 ufromfpxf32 F
+GLIBC_2.27 ufromfpxf32x F
+GLIBC_2.27 ufromfpxf64 F
 GLIBC_2.27 ufromfpxf64x F
 GLIBC_2.27 ufromfpxl F
 GLIBC_2.27 y0 F
 GLIBC_2.27 y0f F
 GLIBC_2.27 y0f128 F
+GLIBC_2.27 y0f32 F
+GLIBC_2.27 y0f32x F
+GLIBC_2.27 y0f64 F
 GLIBC_2.27 y0f64x F
 GLIBC_2.27 y0l F
 GLIBC_2.27 y1 F
 GLIBC_2.27 y1f F
 GLIBC_2.27 y1f128 F
+GLIBC_2.27 y1f32 F
+GLIBC_2.27 y1f32x F
+GLIBC_2.27 y1f64 F
 GLIBC_2.27 y1f64x F
 GLIBC_2.27 y1l F
 GLIBC_2.27 yn F
 GLIBC_2.27 ynf F
 GLIBC_2.27 ynf128 F
+GLIBC_2.27 ynf32 F
+GLIBC_2.27 ynf32x F
+GLIBC_2.27 ynf64 F
 GLIBC_2.27 ynf64x F
 GLIBC_2.27 ynl F
diff --git a/sysdeps/unix/sysv/linux/aarch64/ilp32/libpthread.abilist b/sysdeps/unix/sysv/linux/aarch64/ilp32/libpthread.abilist
index a36bdb8c7a..fcc14ed7a3 100644
--- a/sysdeps/unix/sysv/linux/aarch64/ilp32/libpthread.abilist
+++ b/sysdeps/unix/sysv/linux/aarch64/ilp32/libpthread.abilist
@@ -169,7 +169,6 @@ GLIBC_2.27 pthread_rwlockattr_getpshared F
 GLIBC_2.27 pthread_rwlockattr_init F
 GLIBC_2.27 pthread_rwlockattr_setkind_np F
 GLIBC_2.27 pthread_rwlockattr_setpshared F
-GLIBC_2.27 pthread_self F
 GLIBC_2.27 pthread_setaffinity_np F
 GLIBC_2.27 pthread_setattr_default_np F
 GLIBC_2.27 pthread_setcancelstate F
-- 
2.11.1

